Archive/Imo/Imo1987Q1.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+
+! This file was ported from Lean 3 source module imo.imo1987_q1
+! leanprover-community/mathlib commit 5f25c089cb34db4db112556f23c50d12da81b297
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Fintype.BigOperators
+import Mathlib.Data.Fintype.Perm
+import Mathlib.Data.Fintype.Prod
+import Mathlib.Dynamics.FixedPoints.Basic
+
+/-!
+# Formalization of IMO 1987, Q1
+
+Let $p_{n, k}$ be the number of permutations of a set of cardinality `n ≥ 1` that fix exactly `k`
+elements. Prove that $∑_{k=0}^n k p_{n,k}=n!$.
+
+To prove this identity, we show that both sides are equal to the cardinality of the set
+`{(x : α, σ : perm α) | σ x = x}`, regrouping by `card (fixed_points σ)` for the left hand side and
+by `x` for the right hand side.
+
+The original problem assumes `n ≥ 1`. It turns out that a version with `n * (n - 1)!` in the RHS
+holds true for `n = 0` as well, so we first prove it, then deduce the original version in the case
+`n ≥ 1`. -/
+
+
+variable (α : Type _) [Fintype α] [DecidableEq α]
+
+open scoped BigOperators Nat
+
+open Equiv Fintype Function
+
+open Finset (range sum_const)
+
+open Set (Iic)
+
+namespace Imo1987Q1
```

Comment:
Inconsistently namespaced: another file you did all caps. Let's pick one. :-) I suggest this one.